Content.Server/Anomaly/Effects/IceAnomalySystem.cs

```diff
@@ -0,0 +1,154 @@
+                : new(_mapManager.GetMapEntityId(mapPos.MapId), mapPos.Position);
+
+        var ent = Spawn(component.ProjectilePrototype, spawnCoords);
+        var direction = targetCoords.ToMapPos(_entman, _xform) - mapPos.Position;
+
+        if (!TryComp<ProjectileComponent>(ent, out var comp))
+            return;
+
+        foreach (var key in component.ProjectileDamage.DamageDict.Keys)
+        {
+            comp.Damage.DamageDict[key] = component.ProjectileDamage.DamageDict[key] * severity;
+        }
+
+        _gunSystem.ShootProjectile(ent, direction, Vector2.Zero, uid, component.MaxProjectileSpeed * severity);
+    }
+
+    private void OnSupercritical(EntityUid uid, IceAnomalyComponent component, ref AnomalySupercriticalEvent args)
+    {
+        var xform = Transform(uid);
+        var grid = xform.GridUid;
+        var map = xform.MapUid;
+
+        var indices = _xform.GetGridOrMapTilePosition(uid, xform);
+        var mixture = _atmosphere.GetTileMixture(grid, map, indices, true);
+
+        _boom.QueueExplosion(
+            uid,
+            component.ExplosionPrototype,
+            component.TotalIntensity,
+            component.Dropoff,
+            component.MaxTileIntensity
+        );
+
+        ShootProjectilesAtEntites(uid, component, 1.0f, 1.0f);
+
+        if (mixture == null)
+            return;
+        mixture.AdjustMoles(component.SupercriticalGas, component.SupercriticalMoleAmount);
+        if (grid is { })
+        {
+            foreach (var ind in _atmosphere.GetAdjacentTiles(grid.Value, indices))
+            {
+                var mix = _atmosphere.GetTileMixture(grid, map, ind, true);
+                if (mix is not { })
+                    continue;
+
+                mix.AdjustMoles(component.SupercriticalGas, component.SupercriticalMoleAmount);
+                mix.Temperature += component.FreezeZoneExposeTemperature;
+                _atmosphere.HotspotExpose(grid.Value, indices, component.FreezeZoneExposeTemperature, mix.Volume, uid, true);
+            }
+        }
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<IceAnomalyComponent, AnomalyComponent, TransformComponent>();
+        while (query.MoveNext(out var ent, out var ice, out var anom, out var xform))
+        {
+            var grid = xform.GridUid;
+            var map = xform.MapUid;
+            var indices = _xform.GetGridOrMapTilePosition(ent, xform);
+            var mixture = _atmosphere.GetTileMixture(grid, map, indices, true);
+            if (mixture is { })
+            {
+                mixture.Temperature += ice.ChillPerSecond * anom.Severity * frameTime;
+            }
+
+            if (grid != null && anom.Severity > ice.AnomalyFreezeZoneThreshold)
+            {
+                _atmosphere.HotspotExpose(grid.Value, indices, ice.FreezeZoneExposeTemperature, ice.AnomalyFreezeZoneThreshold, ent, true);
```

Comment:
does exposing it at a negative temp actually reduce the temp? i genuinely don't know. Just assure me that you tested that this does in fact work.

Reply:
If you look at the end of the second video I show the temp of the test chamber being under -90 degrees celsius